doc/xgboost_on_sqlflow.md

```diff
@@ -0,0 +1,139 @@
+# xgboost on sqlflow
+
+## XGBoost 
+
+Gradient boosting machine (GBM) is a widely used (supervised) machine learning method, 
+which trains a bunch of weak learners, typically decision trees, 
+in a gradual, additive and sequential manner. 
+A lot of winning solutions of data mining and machine learning challenges, 
+such as : Kaggle, KDD cup, are based on GBM or related techniques.
+
+XGBoost (https://xgboost.ai/) is an optimized distributed gradient boosting library designed to be highly efficient, 
+flexible and portable, which is often regarded as one of the best GBM frameworks.
+
+
+## xgboost on sqlflow via ant-xgboost
+   
+### Overview
+
+We use [ant-xgboost](https://github.com/alipay/ant-xgboost) as backend,
+which is forked from [xgboost](https://github.com/dmlc/xgboost).
+_Ant-xgboost_ is nearly same as original xgboost, 
+except some improvements to make xgboost easier to use, 
+such as better early stopping strategy, parameter checking, and end to end launcher.
+
+### User Experience
+    
+In terms of sqlflow users, _xgboost_ is an alternative _Estimator_ like _Tensorflow Estimators_. 
+Working with xgboost is quite similar to working with Tensorflow estimators; 
+just change `TRAIN DNNClassifier` into `TRAIN XGBoostEstimator`. 
+In addition, some xgboost specific parameters are required, 
+which can be configured in the same way as _Tensorflow_ parameters. 
+
+Below is a demo about training/predicting via xgboost :
+
+```sql
+// sample clause of train
+select 
+    c1, c2, c3, c4, c5 as class
+from kaggle_credit_fraud_training_data
+TRAIN XGBoostEstimator
+WITH
+  booster = "gbtree"
+  objective = "logistic:binary"
+  eval_metric = "auc"
+  train_eval_ratio = 0.8
+COLUMN
+  NUMERIC(c1) as fc1,
+  BUCKETIZED(fc1, [0, 10, 100]) as fc2,
+  CROSS([fc1, fc2, c3]) as fc3,
```

Comment:
Similarly for `CROSS` feature. Looks like cross feature is equivalent to a multi-level tree, I feel a tree model is able to learn the cross.

Reply:
Agreed tree can automatically learn feature interaction, although empirical study often finds adding human-crafted interactions can further improve accuracy.

Reply:
Cool, let's take `CROSS` off since AntXgBoost is aiming for full AutoML.